packages/authenticator/amplify_authenticator/example/lib/main.dart

```diff
@@ -93,38 +92,78 @@ class _MyAppState extends State<MyApp> {
     // the user is signed in, the Authenticator will use your MaterialApp's
     // navigator to show the correct screen.
     return Authenticator(
-      stringResolver: stringResolver,
-      onException: (exception) {
-        print('[ERROR]: $exception');
+      authenticatorBuilder: (BuildContext context, AuthenticatorState state) {
+        switch (state.currentStep) {
+          case AuthenticatorStep.signIn:
+            return CustomScaffold(
+              state: state,
+              // A prebuilt Sign In form from amplify_authenticator
+              body: SignInForm(),
+              // A custom footer with a button to take the user to sign up
+              footer: Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  const Text('Don\'t have an account?'),
+                  TextButton(
+                    onPressed: () => state.changeStep(
+                      AuthenticatorStep.signUp,
+                    ),
+                    child: const Text('Sign Up'),
+                  ),
+                ],
+              ),
+            );
+          case AuthenticatorStep.signUp:
+            // fictional custom social provider
+            const myAuthProvider = AuthProvider.custom(
+              'heartr',
+              displayName: 'Heartr (fictional)',
+            );
+            const heartIcon = Icon(
+              Icons.favorite,
+              color: Colors.pink,
+              size: 24,
+            );
+            return CustomScaffold(
+              state: state,
+              body: const Column(
+                children: [
+                  SizedBox(height: 24),
+                  Text('SocialSignInButtons (with single provider)'),
+                  SocialSignInButtons(),
+                  SizedBox(height: 24),
+                  Text('single SocialSignInButton (with custom provider)'),
+                  SocialSignInButton(
```

Comment:
Ooh this is kinda weird DX...

Reply:
Can we leverage this flag instead: https://github.com/aws-amplify/amplify-flutter/blob/b3c039213d0cee7c2fe035e42a1f88db24b51e4c/packages/authenticator/amplify_authenticator/lib/src/widgets/form.dart#L400